### PR TITLE
Toilet and urinal updates (flushing dead fish included)

### LINKED ISSUE #81903: Urinals can be used to permanently destroy items (also infinite urinal cake)
## Issue Summary

OK so like you know how you can crowbar a toilet and put something in it? And if you deconstruct the toilet the item drops?
Well you can do the same with urinals, but instead you need to screwdriver it and take the soap out, but it doesn't actually drop the item.

image

Urinals are also an infinite source of urinal cakes cause every time you destroy it and place it back on the wall, it makes a new one to take out.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

I split the water closets file into a folder for all the different structures and also did a few other things
1. Remakes https://github.com/tgstation/tgstation/pull/81914
2. Reworks toilets, it’s now Alt Click to open/close the cover, left click takes things out of the toilet and cistern, right click flushes
3. You can now put dead fish in the toilet to flush it
4. Adds context tips to toilets. 

## Why It's Good For The Game

When I was a 7 year old boy, I had a friend goldfish named Jeremy. He was my greatest friend until one day when my brother told me it’s for babies and flushed him alive down the toilet, and I’ve never seen it again. 

Closes https://github.com/tgstation/tgstation/issues/81903
Lets you flush down people’s fish in the toilet, and re-experience their loss. 

This was inspired by my lawyer fish PR, to add things more on the non-fishing side of fish.

## Changelog

:cl:
add: You can now flush toilets. You can also put fish in the toilet. And flush them.
fix: Urinals can no longer be used to delete items.
fix: Urinals no longer grant infinite urinal cakes. 
/:cl:

### CODE REVIEW

**Comment by the-orange-cow** [comment]
Are you able to give people swirlies?

**Comment by Time-Green** [comment]
> Are you able to give people swirlies?

You’re not gonna believe this

File: code/game/objects/structures/curtains.dm [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	desc = "Contains less than 1% mercury."
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "bathroom-open"
+	color = "#ACD1E9" //Default color, didn't bother hardcoding other colors, mappers can and should easily change it.
+	alpha = 200 //Mappers can also just set this to 255 if they want curtains that can't be seen through
+	layer = SIGN_LAYER
+	anchored = TRUE
+	opacity = FALSE
+	density = FALSE
+	var/icon_type = "bathroom"//used in making the icon state
```
Comment (GoldenAlpharex) [inline code/game/objects/structures/curtains.dm:15]:
```suggestion
	/// used in making the icon state
	var/icon_type = "bathroom"
```

File: code/game/objects/items/clown_items.dm [hunk 2]
```diff
@@ -233,6 +233,22 @@
...
 	COOLDOWN_START(src, golden_horn_cooldown, 1 SECONDS)
 
+/obj/item/bikehorn/rubberducky/plasticducky
+	name = "plastic ducky"
+	desc = "It's a cheap plastic knockoff of a loveable bathtime toy."
+	custom_materials = list(/datum/material/plastic = HALF_SHEET_MATERIAL_AMOUNT)
+
+/obj/item/bikehorn/rubberducky
+	name = "rubber ducky"
+	desc = "Rubber ducky you're so fine, you make bathtime lots of fuuun. Rubber ducky I'm awfully fooooond of yooooouuuu~" //thanks doohl
```
Comment (GoldenAlpharex) [inline code/game/objects/items/clown_items.dm:243]:
```suggestion
	desc = "Rubber ducky you're so fine, you make bathtime lots of fuuun. Rubber ducky I'm awfully fooooond of yooooouuuu.~" //thanks doohl
```

Comment (JohnFulpWillard) [reply code/game/objects/items/clown_items.dm:243]: Hey man I only moved this stuff I don't even know if this is right

**Comment by Time-Green** [comment]
i think it’d also be cool if there was an achievement for getting swirlied to death? something like “a bowl’d new world” would be funny

**Comment by norsvenska** [comment]
> i think it’d also be cool if there was an achievement for getting swirlied to death? something like “a bowl’d new world” would be funny

you're not gonna believe this

**Pull request merged**